cli/electron.ts

```diff
@@ -0,0 +1,280 @@
+"use strict";
+
+import * as p from "./commandparser";
+import * as fs from "fs";
+import * as nodeutil from './nodeutil';
+import * as path from "path";
+
+const gulpCmd = process.platform === "win32" ? "gulp.cmd" : "gulp";
+const nodeCmd = process.platform === "win32" ? "node.cmd" : "node";
+const npmCmd = process.platform === "win32" ? "npm.cmd" : "npm";
+
+let pxtElectronPath: string;
+let pxtElectronSrcPath: string;
+let targetProductJson: string;
+let targetId: string;
+let isInit = false;
+let buildOut: string;
+
+function errorOut(msg: string): Promise<void> {
+    console.error(msg);
+    process.exit(1)
+
+    return null;
+}
+
+export function electronAsync(parsed: p.ParsedCommand): Promise<void> {
+    // Ensure there is a subcommand
+    let subcommand = parsed.arguments[0];
+
+    if (!subcommand) {
+        errorOut("Please specify a subcommand");
+    }
+
+    // Validate current target
+    let needsCurrentTarget = (subcommand !== "build" && subcommand !== "dist") || !parsed.flags["release"];
+
+    if (needsCurrentTarget && !fs.existsSync("pxtarget.json") && !fs.existsSync("package.json")) {
+        errorOut("This command requires to be in a valid target directory (pxtarget.json and package.json required)");
+    }
+
+    targetId = JSON.parse(fs.readFileSync("package.json", "utf8")).name;
```

Comment:
It is not enforced. Simply use the helper function to load the target json on start and you'll have access to pxt.appTarget.id